### DISCUSSION TITLE
Every time the server restarts, the user needs to login again. How to prevent that?

### DISCUSSION THREAD

**Discussion by Ankur-Khandelwal** [jaredhanson/passport#862] [Help]
I am using passport js to set up the authentication for my web application. It's in the development phase. Every time I make changes in the backend and restart the server, the user gets logged out. This is not desirable, especially in the development phase. How to prevent that?

**Comment by jaredhanson** [comment]
The default session store for [`express-session`](https://github.com/expressjs/session) is an in-memory store, which as the name implies does not survive server restarts.  In order to have sessions survive restarts, a durable session store is needed.  [`connect-redis`](https://github.com/tj/connect-redis) is commonly used for this purpose, but there are [numerous others](https://github.com/expressjs/session#compatible-session-stores) available depending on your application's requirements.
